apps/api/src/app/integrations/usecases/update-integration/update-integration.usecase.ts

```diff
@@ -0,0 +1,66 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { IntegrationEntity, IntegrationRepository } from '@notifire/dal';
+import { UpdateIntegrationCommand } from './update-integration.command';
+
+@Injectable()
+export class UpdateIntegration {
+  constructor(private integrationRepository: IntegrationRepository) {}
+
+  async execute(command: UpdateIntegrationCommand): Promise<IntegrationEntity> {
+    const existingIntegration = await this.integrationRepository.findById(command.integrationId);
+    if (!existingIntegration) throw new NotFoundException(`Entity with id ${command.integrationId} not found`);
+
+    const updatePayload: Partial<IntegrationEntity> = {};
+    if (command.providerId) {
+      updatePayload.providerId = command.providerId;
+    }
+
+    if (command.channel) {
+      updatePayload.channel = command.channel;
+    }
+
+    if (command.active) {
+      updatePayload.active = command.active;
+      await this.deactivatedOtherActiveChannels(command);
```

Comment:
I would personally move this to after the update function, just in case the update will throw, we don't have transactions here but still might be a good idea to do it after.

Reply:
Good thinking!